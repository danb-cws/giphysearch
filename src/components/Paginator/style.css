.Paginator {
  width: var(--pagination-button-width);
  padding: 0;
  margin: 0;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  color: var(--control-color);
  background-color: var(--control-bg-color);
  text-transform: uppercase;
  font-size: 2rem;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
}
.Paginator:hover {
  color: var(--control-color-hover);
  background-color: var(--control-bg-color-hover);
}
.Paginator__disabled,
.Paginator__disabled:hover {
  color: var(--control-disabled-color);
  background-color: var(--control-disabled-bg-color);
  cursor: default;
}
.Paginator:focus {
  outline: none;
}
.Paginator__label {
  writing-mode: vertical-rl;
}
.Paginator__label-fwd {
  transform: rotate(180deg);
}

/* fix positioning on ios */
@supports (-webkit-overflow-scrolling: touch) {
  .Paginator__label {
    margin-left: -1.45em;
  }
  .Paginator__label-fwd {
    margin-left: .2em;
  }
}
