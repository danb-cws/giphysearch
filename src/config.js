export const GIPHY_API_KEY = 'dc6zaTOxFJmzC';
export const GIPHY_ENDPOINT = 'https://api.giphy.com/v1/gifs/search?q=';
export const MAX_RESULTS = 5;
export const default_term = 'grumpy cat';