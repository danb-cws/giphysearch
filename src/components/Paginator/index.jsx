import React from "react";
import * as config from "../../config";
import "./style.css";

const Paginator = props => {
  let isDisabled;
  isDisabled =
    (props.dir === -1 && props.resultsPageIndex === 0) ||
    (props.dir === 1 &&
      props.resultsPageIndex + 1 >=
        props.totalResults / config.RESULTS_PER_PAGE);
  return (
    <button
      className={"Paginator" + (isDisabled ? " Paginator__disabled" : "")}
      disabled={isDisabled}
      onClick={() => {
        props.onPaginate(props.dir);
      }}
      tabIndex={props.tabindex}
    >
      <span
        className={
          "Paginator__label" + (props.dir === 1 ? " Paginator__label-fwd" : "")
        }
      >
        {props.children}
      </span>
    </button>
  );
};
export default Paginator;
