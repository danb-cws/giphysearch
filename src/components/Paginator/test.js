import React from "react";
import { shallow } from "enzyme";
import Paginator from "./";
import * as config from "config";

const paginatorClick = jest.fn();

const el = shallow(
  <Paginator
    onPaginate={paginatorClick}
    totalResults={42}
    resultsPageIndex={1}
    dir={1}
  >
    Next
  </Paginator>
);

const elDisabledNext = shallow(
  <Paginator
    onPaginate={paginatorClick}
    totalResults={config.RESULTS_PER_PAGE}
    resultsPageIndex={1}
    dir={1}
  >
    Next
  </Paginator>
);

const elFirstPage = shallow(
  <Paginator
    onPaginate={paginatorClick}
    totalResults={config.RESULTS_PER_PAGE}
    resultsPageIndex={0}
    dir={-1}
  >
    Prev
  </Paginator>
);

const elSecondPage = shallow(
  <Paginator
    onPaginate={paginatorClick}
    totalResults={config.RESULTS_PER_PAGE * 2}
    resultsPageIndex={1}
    dir={-1}
  >
    Prev
  </Paginator>
);

describe("Paginator tests", () => {
  it("Renders the Paginator component", () => {
    expect(el.exists()).toBe(true);
  });
  it("Matches snapshot", () => {
    expect(el).toMatchSnapshot();
  });
  it("responds to click", () => {
    el.simulate("click", { paginatorClick });
    expect(paginatorClick).toBeCalled();
  });
  it("`Next` Paginator should not have disabled class on paginator if enough results for another page", () => {
    expect(el.hasClass("Paginator__disabled")).toBe(false);
  });
  it("`Next` Paginator should have disabled class on  paginator if not enough results for another page", () => {
    expect(elDisabledNext.hasClass("Paginator__disabled")).toBe(true);
  });
  it("`Prev` Paginator should not have disabled class on paginator if not on first page", () => {
    expect(elSecondPage.hasClass("Paginator__disabled")).toBe(false);
  });
  it("`Prev` Paginator should have disabled class on paginator if on first page", () => {
    expect(elFirstPage.hasClass("Paginator__disabled")).toBe(true);
  });
});
