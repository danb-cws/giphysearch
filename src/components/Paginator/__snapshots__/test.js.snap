// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator tests Matches snapshot 1`] = `
<button
  className="Paginator"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="Paginator__label Paginator__label--fwd"
  >
    Next
  </span>
</button>
`;
